### PR TITLE
feat(web): create example routes for csr

### LINKED ISSUE #184: Create example routes with react router
# 🧱 Description

We should create some routes in our frontend application to get familiar with react router.
Make sure to read the [React Router Docs](https://reactrouter.com/en/main/start/tutorial)

## 📋 Acceptance Criteria

- [ ] Our web application has working routes, with buttons to change routes.

### PR DESCRIPTION
**Pull request by hks1444**
This is a basic example of routing. Any suggestions about the code are welcomed. Feel free to change the code and UI. It resolves #184 .

### CODE REVIEW

File: web/src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,106 @@
...
+import {
+  createBrowserRouter,
+  RouterProvider,
+  Outlet
+} from "react-router-dom";
+import img from "../public/zenith-logo.svg";
 
+interface NumberButtonProps {
+  digit: number,
+}
```
Comment (umitcan07) [inline web/src/App.tsx:15]: let's use type instead.
types in TS are considered to be more reliable than interfaces.

File: web/src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,106 @@
...
 import { theme } from "./theme";
+import * as ReactDOM from "react-dom/client";
+import {
+  createBrowserRouter,
+  RouterProvider,
+  Outlet
+} from "react-router-dom";
+import img from "../public/zenith-logo.svg";
 
+interface NumberButtonProps {
```
Comment (umitcan07) [inline web/src/App.tsx:13]:
```suggestion
type NumberButtonProps = {
```

File: web/src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,106 @@
...
+  digit: number,
+}
+
+const Buttons = function (props: NumberButtonProps) {
```
Comment (umitcan07) [inline web/src/App.tsx:17]: can you use arrow function?

const Buttons = ({digit}: NumberButtonProps)

If you also use restructuring, you will not need to type

`props.digit` everytime, just `digit` will suffice

File: web/src/App.tsx [hunk 4]
```diff
@@ -1,14 +1,106 @@
...
+    ],
+  },
+]);
 export default function App() {
   return (
     <MantineProvider theme={theme}>
-      <Container>
-        <img src="./zenith-logo.svg" alt="Zenith Logo" />
-      </Container>
+      <RouterProvider router={router} />
```
Comment (umitcan07) [inline web/src/App.tsx:103]: nice to have added router provider but can you keep Container here?

**Review by umitcan07 — CHANGES_REQUESTED** [review]
Can you make the aforementioned changes?

**Comment by hks1444** [comment]
I made another commit can you check @umitcan07

**Comment by umitcan07** [comment]
Thanks for resolving conflicts gracefully. Good job

**Review by umitcan07 — APPROVED** [review]
Good first PR on web!

**Pull request merged**